### ISSUE TITLE
perl-5.8.8 compatibility

### ISSUE THREAD

**Issue by ferki** [RexOps/Rex#578]
## TL:DR version

We continue to support running Rex on CentOS/RHEL 5, but general usage with perl-5.8.8 is not supported (yet?).
## Long story

Since CentOS/RHEL 5 is supported by Rex, and those OSes are still shipping perl-5.8.8, we need to ensure that Rex can actually be _run_ on those systems (please note that this does not the same as the ability to remotely _manage_ those OSes).

Unfortunately 5.8.8 has a bug related to the `/m` regexp modifier. For more details, see the following upstream perl bugs:
- [RT#3038](https://rt.perl.org/Public/Bug/Display.html?id=3038)
- [RT#7781](https://rt.perl.org/Public/Bug/Display.html?id=7781)
- [RT#10054](https://rt.perl.org/Public/Bug/Display.html?id=10054)

For general 5.8.8 compatibility, all the current regular expressions that rely on the correct `/m` modifier behaviour needs to be replaced. Proposed [workaround from this Stack Overflow thread](http://stackoverflow.com/questions/11231441/regex-ms-not-behaving-as-expected-perl-5-8-8):
- `qr/^/m` is equivalent to `qr/(?:^|(?<=\n))/`
- `qr/$/m` is equivalent to `qr/(?=\n|\z)/`

Fortunately CentOS/RHEL 5 is not affected by this bug because they backported the upstream fix into `perl-5.8.8-6` (see [Changelog](http://rpm.pbone.net/index.php3/stat/22/idpl/27814152/com/changelog.html)).

**Comment by ferki** [comment]
With the "End of Maintenance Support 2" of CentOS/RHEL 5 on 2017-03-31, we already bumped minimum supported perl version for _running_ Rex to 5.10.1.

The related commit is c4e35d6 as part of #1222.
